crates/swc_ecma_minifier/src/compress/optimize/params.rs

```diff
@@ -0,0 +1,371 @@
+            let mut inlinable = true;
+
+            for call_site in call_sites {
+                let arg_value = if param_idx < call_site.len() {
+                    call_site[param_idx].clone()
+                } else {
+                    // Implicit undefined - use unresolved context
+                    Some(Box::new(Expr::Ident(Ident::new(
+                        "undefined".into(),
+                        DUMMY_SP,
+                        self.ctx.expr_ctx.unresolved_ctxt,
+                    ))))
+                };
+
+                let arg_value = match arg_value {
+                    Some(v) => v,
+                    None => {
+                        inlinable = false;
+                        break;
+                    }
+                };
+
+                // Check if this is a safe, constant value to inline
+                if !self.is_safe_constant_for_param_inline(&arg_value) {
+                    inlinable = false;
+                    break;
+                }
+
+                match &common_value {
+                    None => {
+                        common_value = Some(arg_value);
+                    }
+                    Some(existing) => {
+                        if !self.expr_eq(existing, &arg_value) {
+                            inlinable = false;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // If all call sites pass the same constant value, mark for inlining
+            if inlinable {
+                if let Some(value) = common_value {
+                    params_to_inline.push((param_idx, value, is_reassigned));
+                }
+            }
+        }
+
+        // Apply the parameter inlining transformation
+        if !params_to_inline.is_empty() {
+            self.apply_param_inlining(f, &params_to_inline, fn_id);
+        }
+    }
+
+    /// Apply parameter inlining transformation to a function.
+    fn apply_param_inlining(
+        &mut self,
+        f: &mut Function,
+        params_to_inline: &[(usize, Box<Expr>, bool)],
+        fn_id: &Id,
+    ) {
+        // Sort in reverse order to remove from the end first
+        let mut sorted_params = params_to_inline.to_vec();
+        sorted_params.sort_by(|a, b| b.0.cmp(&a.0));
+
+        // Collect variable declarations to add at the beginning of function body
+        let mut var_decls = Vec::new();
+
+        // Track which parameter indices are being inlined
+        let mut inlined_indices = FxHashSet::default();
+
+        for (param_idx, value, is_reassigned) in &sorted_params {
+            inlined_indices.insert(*param_idx);
+
+            if let Some(param) = f.params.get(*param_idx) {
+                if let Pat::Ident(ident) = &param.pat {
+                    // Create variable declaration: const/let paramName = value;
+                    // Use 'let' if the parameter is reassigned, 'const' otherwise
```

Comment:
for simplicity i would just always use `let` and not even bother tracking wither it was reassigned.  This preserves the default semantics of a parameter and should save the analysis.  

I guess unless that would inhibit other passes that want to constant fold the value?